### ISSUE TITLE
What happened to drupal_setcookie()?

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6020]
I was looking at [this change record](https://www.drupal.org/node/3275352), to see if there's anything that may apply to Backdrop for backporting. I also checked [the respective issue for D10](https://www.drupal.org/project/drupal/issues/3150614), and the have also noticed [this D7 issue](https://www.drupal.org/project/drupal/issues/3282349), which we'll eventually need to consider crossporting I am assuming.

Anyway, I looked in our codebase, our api site and change records, but couldn't find anything related to `drupal_setcookie()` nor a respective `backdrop_setcookie()` function. Does anyone know how come?

**Comment by argiepiano** [comment]
You probably saw this already - it looks like `drupal_setcookie()` is a wrapper around PHP's `setcookie()` and is used very sparsely in Drupal. One such place is `drupal_session_commit()`. The parallel function in Backdrop `backdrop_session_commit()` uses `setcookie()`. So my guess is that that wrapper was deemed unnecessary and Backdrop reverted to the PHP function.

**Comment by klonos** [comment]
Thanks @argiepiano, yes, I saw that. Nevertheless, I think that we still need a change record at minimum. Perhaps also a wrapper function in `core/includes/drupal.inc` that calls `setcookie()` and logs a watchdog message.

**Comment by klonos** [comment]
Yes, it was introduced with https://git.drupalcode.org/project/drupal/-/commit/7ba88d9d4f2d38f563c3f63a70f24a91a848e192 in D 7.79 ([issue](https://www.drupal.org/project/drupal/issues/3170525) | [change record](https://www.drupal.org/node/3207213))

> PHP introduced native support for the SameSite cookie attribute in PHP 7.3 so Drupal emulates this functionality for earlier PHP versions.

Sounds like we need to crossport this then, so changing this to a task.

**Comment by klonos** [comment]
...we should also include https://www.drupal.org/project/drupal/issues/3282349 (currently RTBC) and possibly backport https://www.drupal.org/node/3150614 from D10 (default the attribute's value to `Lax`).

**Comment by kiamlaluno** [comment]
Given the linked issues, should not this task also be for sites using PHP 7.3 or higher versions? For what I understand, the default value PHP would set is not necessarily the value site settings would use.
